tests/python_package_test/test_dask.py

```diff
@@ -97,6 +98,57 @@ def test_classifier(output, centers, client, listen_port):
     assert_eq(p1_proba, p2_proba, atol=0.3)
 
 
+@pytest.mark.parametrize('output', data_output)
+@pytest.mark.parametrize('centers', data_centers)
+def test_classifier_pred_contrib(output, centers, client, listen_port):
+    X, y, w, dX, dy, dw = _create_data('classification', output=output, centers=centers)
+
+    dask_classifier = dlgbm.DaskLGBMClassifier(
+        time_out=5,
+        local_listen_port=listen_port,
+        tree_learner='data'
+    )
+    dask_classifier = dask_classifier.fit(dX, dy, sample_weight=dw, client=client)
+    preds_with_contrib = dask_classifier.predict(dX, pred_contrib=True).compute()
+
+    local_classifier = lightgbm.LGBMClassifier()
+    local_classifier.fit(X, y, sample_weight=w)
+    local_preds_with_contrib = local_classifier.predict(X, pred_contrib=True)
+
+    if output == 'scipy_csr_matrix':
+        preds_with_contrib = np.array(preds_with_contrib.todense())
+
+    # shape depends on whether it is binary or multiclass classification
+    num_classes = len(centers)
```

Comment:
Those features are already supported, because `DaskLGBMClassifier` inherits from `lightgbm.sklearn.LGBMClassifier` and `DaskLGBMRegressor` inherits from `lightgbm.sklearn.LGBMRegressor`. It even has a `.booster_`, so you don't have to use `to_local()` if you just want the booster directly.

![image](https://user-images.githubusercontent.com/7608904/105282883-e9191080-5b74-11eb-925d-77e2f75a3263.png)

I can change this test to use `.n_classes_`, that's fine.

Reply:
Wow, awesome indeed! Thanks for clarification!